--- origsrc/libcdio-0.80/lib/driver/device.c	2006-04-03 21:06:13.000000000 -0500
+++ src/libcdio-0.80/lib/driver/device.c	2008-04-24 14:37:53.390625000 -0500
@@ -68,9 +68,9 @@
 const driver_id_t cdio_os_driver = DRIVER_LINUX;
 #elif  HAVE_DARWIN_CDROM
 const driver_id_t cdio_os_driver = DRIVER_OSX;
-#elif  HAVE_DARWIN_SOLARIS
+#elif  HAVE_SOLARIS_CDROM
 const driver_id_t cdio_os_driver = DRIVER_SOLARIS;
-#elif  HAVE_DARWIN_WIN32
+#elif  HAVE_WIN32_CDROM
 const driver_id_t cdio_os_driver = DRIVER_WIN32;
 #else 
 const driver_id_t cdio_os_driver = DRIVER_UNKNOWN;
