--- origsrc/libcdio-0.81/configure.ac	2008-10-27 01:48:44.000000000 -0500
+++ src/libcdio-0.81/configure.ac	2008-12-28 12:30:44.284375000 -0600
@@ -502,6 +502,7 @@
      	AC_DEFINE([HAVE_WIN32_CDROM], [1], 
                   [Define 1 if you have MinGW CD-ROM support])
  	LIBS="$LIBS -lwinmm"
+ 	LT_NO_UNDEFINED="-no-undefined"
   	cd_drivers="${cd_drivers}, MinGW"
      	AC_DEFINE([NEED_TIMEZONEVAR], [1], 
                   [Define 1 if you need timezone defined to get timzone
--- origsrc/libcdio-0.81/lib/cdio++/Makefile.am	2008-10-19 20:25:15.000000000 -0500
+++ src/libcdio-0.81/lib/cdio++/Makefile.am	2008-12-28 12:32:15.128125000 -0600
@@ -50,7 +50,8 @@
 libcdiopp_sources = cdio.cpp devices.cpp
 
 libcdio___la_SOURCES = $(libcdiopp_sources)
-libcdio___la_ldflags = -version-info $(libcdiopp_la_CURRENT):$(libcdiopp_la_REVISION):$(libcdiopp_la_AGE) @LT_NO_UNDEFINED@
+libcdio___la_LDFLAGS = -version-info $(libcdiopp_la_CURRENT):$(libcdiopp_la_REVISION):$(libcdiopp_la_AGE) @LT_NO_UNDEFINED@
+libcdio___la_LIBADD = $(top_builddir)/lib/driver/libcdio.la
 
 libiso9660pp_la_CURRENT = 0
 libiso9660pp_la_REVISION = 0
@@ -59,6 +60,7 @@
 libiso9660pp_sources = iso9660.cpp
 
 libiso9660___la_SOURCES = $(libiso9660pp_sources)
-libiso9660___la_ldflags = -version-info $(libiso9660pp_la_CURRENT):$(libiso9660pp_la_REVISION):$(libiso9660pp_la_AGE) @LT_NO_UNDEFINED@
+libiso9660___la_LDFLAGS = -version-info $(libiso9660pp_la_CURRENT):$(libiso9660pp_la_REVISION):$(libiso9660pp_la_AGE) @LT_NO_UNDEFINED@
+libiso9660___la_LIBADD = $(top_builddir)/lib/iso9660/libiso9660.la
 
 INCLUDES = -I$(top_srcdir)/include/ -I$(top_builddir)/include
--- origsrc/libcdio-0.81/lib/iso9660/Makefile.am	2008-10-19 20:25:15.000000000 -0500
+++ src/libcdio-0.81/lib/iso9660/Makefile.am	2008-12-28 12:36:12.096875000 -0600
@@ -67,7 +67,7 @@
 
 libiso9660_la_LIBADD = @LIBCDIO_LIBS@
 libiso9660_la_ldflags = -version-info $(libiso9660_la_CURRENT):$(libiso9660_la_REVISION):$(libiso9660_la_AGE) @LT_NO_UNDEFINED@
-libiso9660_la_dependencies = libcdio.la
+libiso9660_la_dependencies = $(top_builddir)/lib/driver/libcdio.la
 
 INCLUDES = $(LIBCDIO_CFLAGS)
 
