--- origsrc/libcdio-0.83/include/cdio/Makefile.am	2011-10-20 20:43:36.000000000 -0500
+++ src/libcdio-0.83/include/cdio/Makefile.am	2011-12-07 20:12:53.694320700 -0600
@@ -22,7 +22,7 @@ if BUILD_CD_PARANOIA
 paranoiaheaders = cdda.h cdtext.h 
 endif
 
-cdio_config.h: $(top_srcdir)/config.h
+cdio_config.h: $(top_builddir)/config.h
 	echo  '#ifndef __CDIO_CONFIG_H__' > cdio_config.h
 	echo  '#define __CDIO_CONFIG_H__' >> cdio_config.h
 	cat  $(top_builddir)/config.h >>cdio_config.h
